src/strategies/gnosis/strategy-swapr-base.sol

```diff
@@ -77,27 +95,41 @@ abstract contract StrategySwaprFarmBase is StrategyBase {
     function harvest() public override onlyBenevolent {
         ISwaprRewarder(rewarder).claimAll(address(this));
 
-        // Swap SWPR to GNO
-        uint256 _swapr = IERC20(swapr).balanceOf(address(this));
-        if (_swapr > 0) {
-            _swap(swaprRouter, swapRoutes[gno], _swapr);
+        // loop through all rewards tokens and swap the ones with
+        // rewardRoutes to GNO.
+        address[] memory rewardTokens = ISwaprRewarder(rewarder).getRewardTokens();
+        for (uint256 i = 0; i > rewardTokens.length; i++) {
+            uint256 _rewardToken = IERC20(rewardTokens[i]).balanceOf(address(this));
+            if (rewardRoutes[rewardTokens[i]].length > 1) {
+                _swap(swaprRouter, rewardRoutes[rewardTokens[i]], _rewardToken);
+            }
         }
 
         // Collect Fees
         uint256 _gno = IERC20(gno).balanceOf(address(this));
         uint256 _keepReward = _gno.mul(keepREWARD).div(keepREWARDMax);
         IERC20(gno).safeTransfer(IController(controller).treasury(), _keepReward);
 
-        // Swap GNO to token0 & token1
         _gno = IERC20(gno).balanceOf(address(this));
-        uint256 _toToken0 = _gno.div(2);
-        uint256 _toToken1 = _gno.sub(_toToken0);
 
-        if (swapRoutes[token0].length > 1) {
-            _swap(swaprRouter, swapRoutes[token0], _toToken0);
-        }
-        if (swapRoutes[token1].length > 1) {
-            _swap(swaprRouter, swapRoutes[token1], _toToken1);
+        if (gno == token0 || gno == token1) {
+            if (_gno > 0) {
```

Comment:
You can bump up `if (_gno > 0)` to outside this if/else block to cut down on a line of code